sigmf/sigmffile.py

```diff
@@ -183,7 +216,7 @@ def set_global_info(self, new_global):
         Overwrite the global info with a new dictionary.
         """
         self._validate_dict_in_section(new_global, self.GLOBAL_KEY)
-        self._metadata[self.GLOBAL_KEY] = new_global
+        self._metadata[self.GLOBAL_KEY] = new_global.copy()
```

Comment:
If `new_global` is a dict that is used to create multiple `.sigmf` files, then we probably don't want things like the `sha512` to be reflected in the dict at the level that invoked (for example) `SigMFFile.__init__()`